util/types/time.go

```diff
@@ -437,6 +437,15 @@ func (t *Time) check() error {
 	return nil
 }
 
+// TotalDays returns number of days starting from 0/1/1
+func (t Time) TotalDays() int {
```

Comment:
[`calcDaynr`](https://github.com/pingcap/tidb/blob/master/util/types/mytime.go#L170) function is not exported. 

I test mysql behavior and find month or day can't contain 0:

```
mysql> select datediff('2016-01-02', '2016-00-13');
+--------------------------------------+
| datediff('2016-01-02', '2016-00-13') |
+--------------------------------------+
|                                 NULL |
+--------------------------------------+
```

so I think it should be acceptable to do it like this, if the value doesn't overflow:
```
t1, err := t.GoTime()
duration := t1.Sub(time.Date(0, 1, 1, 0, 0, 0, 0, time.Local))
return duration / (24 * time.Hour)
```

@coocood